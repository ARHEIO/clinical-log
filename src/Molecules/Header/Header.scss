@import '../../styleguide.scss';

.app-header {

  &_top-bar {
    height: 25px;
    background-color: #546A7B;
  }

  &_bottom-bar {
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    width: 100%;
    height: 80px;
    background-color: #87AEB5;

    .container {
      align-items: center;
      display: flex;
      flex-direction: row;
      justify-content: space-between;

      h1 {
        display: inline-block;
        font-family: Quicksand,sans-serif;
        font-size: 1.5em;
        font-stretch: condensed;
        font-weight: 400;
        letter-spacing: 0.1em;
        line-height: 1.1;
        // text-shadow: 1px 1px 10px hsla(0, 0%, 100%, 0.25), -1px -1px 10px hsla(0, 0%, 100%, 0.25);
        text-transform: uppercase;
        color: #393D3F;
        // transition: text-shadow 0.8s cubic-bezier(0.5, 0.18, 0.115, 0.87);
      }

      a {
        min-width: 80px;
        max-width: 150px;
      }

      @media (min-width: 960px) {
        width: 800px;
      }
    }
  }
}
