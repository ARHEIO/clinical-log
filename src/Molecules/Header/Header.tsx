import React, { ReactElement, useContext } from 'react';
import './Header.scss';
import Button from '@material-ui/core/Button';
import { AuthContext } from '../../Store/Store';

const Header = (): ReactElement => {
  const [authContext, setAuthContext] = useContext(AuthContext);

  const logInUser = (): void => {
    setAuthContext({ isAuthed: true });
  };

  return (
    <header className="app-header">
      <div className="app-header_top-bar" />
      <div className="app-header_bottom-bar">
        <div className="container">
          <h1>South Yarra Health and Wellbeing Clinic</h1>
          {authContext.isAuthed
            ? <p>Welcome Chris</p>
            : <Button className="log-in-button" variant="contained" color="primary" href="#/auth" onClick={logInUser}>LOGIN</Button>}
        </div>
      </div>
    </header>
  );
};

export default Header;
