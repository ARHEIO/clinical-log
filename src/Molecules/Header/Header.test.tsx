import React from 'react';
import { render } from '@testing-library/react';
import pretty from 'pretty';
import Header from './Header';

it('renders learn react link', () => {
  const { container } = render(<Header />);
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
