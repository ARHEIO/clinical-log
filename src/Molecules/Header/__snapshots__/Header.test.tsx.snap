// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render when authed 1`] = `
"<header class=\\"app-header\\">
  <div class=\\"app-header_top-bar\\"></div>
  <div class=\\"app-header_bottom-bar\\">
    <div class=\\"container\\">
      <h1>South Yarra Health and Wellbeing Clinic</h1>
      <p>Welcome Chris</p>
    </div>
  </div>
</header>"
`;

exports[`Header should render when unauthed 1`] = `
"<header class=\\"app-header\\">
  <div class=\\"app-header_top-bar\\"></div>
  <div class=\\"app-header_bottom-bar\\">
    <div class=\\"container\\">
      <h1>South Yarra Health and Wellbeing Clinic</h1><a class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained log-in-button MuiButton-containedPrimary\\" tabindex=\\"0\\" aria-disabled=\\"false\\" href=\\"#/auth\\"><span class=\\"MuiButton-label\\">LOGIN</span><span class=\\"MuiTouchRipple-root\\"></span></a>
    </div>
  </div>
</header>"
`;
